### ISSUE TITLE
use_frameworks in podfile causes build error

### ISSUE THREAD

**Issue by mypark** [react-native-community/react-native-maps#2218]
### Is this a bug report?
Yes.
Having a pod install with use_frameworks flag causes build errors with react-navtive-maps.  Looks like this other issue is related too: https://github.com/react-community/react-native-maps/issues/2202

### Have you read the [Installation Instructions](https://github.com/react-community/react-native-maps/blob/master/docs/installation.md)?

Yes I followed the instructions.  I am not running google maps or implementing for android.

### Environment
iOS

"react": "16.3.1",
"react-native": "^0.55.3",
"react-native-maps": "^0.21.0",
"react-viro": "2.6.0",

### Steps to Reproduce

1. npm install react-viro

2. add this to the podfile
  use_frameworks!
  pod 'ViroReact', :path => '../node_modules/react-viro/ios/'
  pod 'ViroKit', :path => '../node_modules/react-viro/ios/dist/ViroRenderer/'

3. pod install

4. Build

### Expected Behavior

It should build without errors

### Actual Behavior

Missing dependency target "React" and a React/RCTView.h' file not found error

### Reproducible Demo

Podfile 

```
platform :ios, '11.0'
target 'MyGreatApp' do
  use_frameworks!
  pod 'ViroReact', :path => '../node_modules/react-viro/ios/'
  pod 'ViroKit', :path => '../node_modules/react-viro/ios/dist/ViroRenderer/'
  # Required by RNFirebase
  pod 'Firebase/Core'
  pod 'Firebase/Auth'
  pod 'Firebase/Firestore'
  pod 'Firebase/Messaging'
  pod 'Fabric', '~> 1.7.5'
  pod 'Crashlytics', '~> 3.10.1'
  pod 'Firebase/Database'
  pod 'Firebase/RemoteConfig'
  pod 'Firebase/Storage'
  pod 'Firebase/Performance'
  pod 'Firebase/DynamicLinks'

  rn_path = '../node_modules/react-native'
  rn_maps_path = '../node_modules/react-native-maps'

  # See http://facebook.github.io/react-native/docs/integration-with-existing-apps.html#configuring-cocoapods-dependencies
  pod 'yoga', path: "#{rn_path}/ReactCommon/yoga/yoga.podspec"
  pod 'React', path: rn_path, subspecs: [
    'Core',
    'CxxBridge',
    'DevSupport',
    'RCTActionSheet',
    'RCTAnimation',
    'RCTGeolocation',
    'RCTImage',
    'RCTLinkingIOS',
    'RCTNetwork',
    'RCTSettings',
    'RCTText',
    'RCTVibration',
    'RCTWebSocket',
  ]



  # React Native third party dependencies podspecs
  pod 'DoubleConversion', :podspec => "#{rn_path}/third-party-podspecs/DoubleConversion.podspec"
  pod 'glog', :podspec => "#{rn_path}/third-party-podspecs/glog.podspec"
  pod 'Folly', :podspec => "#{rn_path}/third-party-podspecs/Folly.podspec"

  # react-native-maps dependencies
  pod 'react-native-maps', path: rn_maps_path
end

pre_install do |installer|
# workaround for CocoaPods/CocoaPods#3289
  Pod::Installer::Xcode::TargetValidator.send(:define_method, :verify_no_static_framework_transitive_dependencies) {}
end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    if target.name == 'react-native-google-maps'
      target.build_configurations.each do |config|
        config.build_settings['CLANG_ENABLE_MODULES'] = 'No'
      end
    end
    if target.name == "React"
      target.remove_from_project
    end
  end
end



```

**Comment by smooJitter** [comment]
All ways a problem installing this package.  I don’t think your pod file should have the react hook ups.  They are redundant.

**Comment by mypark** [comment]
I tried removing it but it still gives me build errors

**Comment by rborn** [comment]
`use_frameworks` is not supported by this module sadly.

**Comment by aksonov** [comment]
@rborn Is it official? `KissXML` framework (used by XMPPFramework) doesn't work without `use_frameworks` now. Does it mean that we can't use XMPPFramework AND react-native-maps for the same project?

**Comment by rborn** [comment]
@aksonov I didn't find a way, my objC is not as good so... 😞

**Comment by mypark** [comment]
The only way I got it to work was to not use the podfile for
react-native-maps and link the project manually and remove the google maps
project files so that it was using the native mapkit.

**Comment by olegserov** [comment]
@mypark Would you mind sharing how exactly did you added it manually? Having same issue and I've spend a lot of time trying to fix it.

**Comment by mypark** [comment]
@olegserov after npm installing, I went into XCode, right clicked on libraries folder and added the project by pointing to the react-native-maps project file in the node_modules folder.  Then I dragged the library to link binary section.  The instructions are here in the "link manually" part.  (However, Don't run react-native link)

https://facebook.github.io/react-native/docs/linking-libraries-ios.html

After I added it, I went and removed the google maps folder from the library project to get rid of the build errors.

**Comment by olegserov** [comment]
Please see this solution that works without manual linking:
https://github.com/react-community/react-native-maps/issues/2324

**Comment by muhammadsr** [comment]
Here are some notes to what I did to make it work on iOS. I ran into two issues that were preventing me from having the map library work.

1. I had the React and ReactNativeNavigation Frameworks  linked manually in our project rather than using `Podfile`. However, when we introduced the `react-native-maps` which is requiring me to use `Podfile` to install it, it was having issues referencing the React framework since it was installed manually. If I get rid of the manual linking and use `Podfile` to link React and ReactNativeNavigation, that will solve this issue.

2. The `react-native-maps` does not work properly when you use `use_frameworks!`. So, I had to install this plugin `cocoapods-fix-react-native`. 

Thanks to @olegserov for his solution here https://github.com/react-community/react-native-maps/issues/2324
